crud/src/main/java/org/openmbee/mms/crud/services/DefaultBranchService.java

```diff
@@ -145,18 +152,26 @@ public RefJson createBranch(String projectId, RefJson branch) {
             b.setParentRefId(Constants.MASTER_BRANCH);
         }
 
-        //This service cannot create branches from historic versions
-        if (branch.getParentCommitId() != null) {
-            throw new BadRequestException("Internal Error: Invalid branch creation logic.");
-        }
-
         Optional<Branch> refOption = branchRepository.findByBranchId(b.getParentRefId());
         if (refOption.isPresent()) {
-            Optional<Commit> parentCommit = commitRepository.findLatestByRef(refOption.get());
-            parentCommit.ifPresent(parent -> {
-                b.setParentCommit(parent.getId());
-                branch.setParentCommitId(parent.getCommitId()); //commit id is same as its docId
-            });
+            if(branch.getParentCommitId() != null){
+                Optional<Commit> commitRequestId = commitRepository.findByCommitId(branch.getParentCommitId());
+                commitRequestId.ifPresentOrElse(commit -> {
+                    Optional<Commit> parentCommit = commitRepository.findByRefAndTimestamp(refOption.get(), commit.getTimestamp());
+                    parentCommit.ifPresent(parent -> {
+                        b.setParentCommit(parent.getId());
+                        branch.setParentCommitId(parent.getCommitId());
+                    });
+                },
```

Comment:
This method was updated so that if a RefJson has a parentCommitId, we don't set the new branch's parentCommitId to the the latest from the parent ref.